src/pages/settings/PasswordPage.js

```diff
@@ -43,50 +43,45 @@ const propTypes = {
 const defaultProps = {
     account: {},
 };
-class PasswordPage extends Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            currentPassword: '',
-            newPassword: '',
-            errors: {
-                currentPassword: false,
-                newPassword: false,
-                newPasswordSameAsOld: false,
-            },
-        };
-
-        this.submit = this.submit.bind(this);
-        this.getErrorText = this.getErrorText.bind(this);
-        this.validate = this.validate.bind(this);
-        this.clearErrorAndSetValue = this.clearErrorAndSetValue.bind(this);
 
-        this.errorKeysMap = {
-            currentPassword: 'passwordPage.errors.currentPassword',
-            newPasswordSameAsOld: 'passwordPage.errors.newPasswordSameAsOld',
-            newPassword: 'passwordPage.errors.newPassword',
-        };
-    }
+const errorKeysMap = {
+    currentPassword: 'passwordPage.errors.currentPassword',
+    newPasswordSameAsOld: 'passwordPage.errors.newPasswordSameAsOld',
+    newPassword: 'passwordPage.errors.newPassword',
+};
 
-    componentWillUnmount() {
-        Session.clearAccountMessages();
-    }
+function PasswordPage(props) {
+    const [passwordFields, setPasswordFields] = useState({
+        currentPassword: '',
+        newPassword: '',
+    });
+    const [errors, setErrors] = useState({
+        currentPassword: false,
+        newPassword: false,
+        newPasswordSameAsOld: false,
+    });
+
+    const currentPasswordInputRef = useRef(null);
+
+    useEffect(
+        () => () => {
+            Session.clearAccountMessages();
+        },
+        [],
+    );
```

Comment:
I think you can make this one-line and it still falls within our style guidelines

```Javascript 
  useEffect(() => () => Session.clearAccountMessages(),[]);
```